### ISSUE TITLE
TRTorch on torchvision ResNet152

### ISSUE THREAD

**Issue by esghif** [NVIDIA/TRTorch#84]
Hi, 

I tried the script below and get error messages and a segmentation fault. Is this a bug or am I doing it wrong?

`import copy
import itertools
import logging
import numpy as np
import os
import sys

import torch
import torchvision.models
import trtorch

def torchvision_benchmark():    
    os.environ["CUDA_DEVICE_ORDER"]="PCI_BUS_ID"
    os.environ["CUDA_VISIBLE_DEVICES"]="0,1,2"

    B, C, H, W = 1, 3, 224, 224

    pytorch_model = torchvision.models.resnet152(pretrained=True)
    print(f"Started torch.jit.script() ...", end='')
    torchscript_model = torch.jit.script(copy.deepcopy(pytorch_model))
    print(f" done.")

    compile_settings = {
        "input_shapes": [[1, 3, 224, 224]],
        "op_precision": torch.float
    }

    for i, (k, v) in enumerate(torchscript_model.named_parameters()):
        print(k, v.shape)
        if i > 10:
            break  

    _ = torch.jit.script(copy.deepcopy(pytorch_model).eval())
    graph_lines = str(_.inlined_graph).split('\n')
    ls, le = 0, 20
    for l in graph_lines[ls:le]:
        print(l)
    print(f"Started trtorch.compile() ...", end='')
    trt_ts_module = trtorch.compile(_, compile_settings)
    print(f" done.")
`

I get following results:

> Started torch.jit.script() ... done.
conv1.weight torch.Size([64, 3, 7, 7])
bn1.weight torch.Size([64])
bn1.bias torch.Size([64])
layer1.0.conv1.weight torch.Size([64, 64, 1, 1])
layer1.0.bn1.weight torch.Size([64])
layer1.0.bn1.bias torch.Size([64])
layer1.0.conv2.weight torch.Size([64, 64, 3, 3])
layer1.0.bn2.weight torch.Size([64])
layer1.0.bn2.bias torch.Size([64])
layer1.0.conv3.weight torch.Size([256, 64, 1, 1])
layer1.0.bn3.weight torch.Size([256])
layer1.0.bn3.bias torch.Size([256])
graph(%self : __torch__.torchvision.models.resnet.ResNet,
      %x.1 : Tensor):
  %3 : int = prim::Constant[value=-1]()
  %4 : int = prim::Constant[value=1]() # /opt/conda/lib/python3.7/site-packages/torchvision-0.7.0a0+34810c0-py3.7-linux-x86_64.egg/torchvision/models/resnet.py:214:29
  %5 : __torch__.torch.nn.modules.conv.Conv2d = prim::GetAttr[name="conv1"](%self)
  %6 : Tensor = prim::GetAttr[name="weight"](%5)
  %7 : int = prim::Constant[value=3]() # /opt/conda/lib/python3.7/site-packages/torch/nn/modules/conv.py:346:24
  %8 : int = prim::Constant[value=1]() # /opt/conda/lib/python3.7/site-packages/torch/nn/modules/conv.py:344:38
  %9 : int = prim::Constant[value=2]() # /opt/conda/lib/python3.7/site-packages/torch/nn/modules/conv.py:343:47
  %10 : Tensor? = prim::GetAttr[name="bias"](%5)
  %11 : int[] = prim::ListConstruct(%9, %9)
  %12 : int[] = prim::ListConstruct(%7, %7)
  %13 : int[] = prim::ListConstruct(%8, %8)
  %x.3 : Tensor = aten::conv2d(%x.1, %6, %10, %11, %12, %13, %8) # /opt/conda/lib/python3.7/site-packages/torch/nn/modules/conv.py:345:15
  %15 : __torch__.torch.nn.modules.batchnorm.BatchNorm2d = prim::GetAttr[name="bn1"](%self)
  %16 : Function = prim::Constant[name="batch_norm"]()
  %17 : float = prim::Constant[value=1.0000000000000001e-05]() # /opt/conda/lib/python3.7/site-packages/torch/nn/modules/batchnorm.py:106:40
  %18 : bool = prim::Constant[value=0]() # /opt/conda/lib/python3.7/site-packages/torch/nn/modules/batchnorm.py:94:11
  %19 : bool = prim::Constant[value=1]() # /opt/conda/lib/python3.7/site-packages/torch/nn/modules/batchnorm.py:94:29
  %exponential_average_factor.126 : float = prim::Constant[value=0.10000000000000001]() # /opt/conda/lib/python3.7/site-packages/torch/nn/modules/batchnorm.py:92:41
ERROR: [TRTorch Conversion Context] - %791 : Tensor = aten::_convolution(%x.1, %self.conv1.weight, %self.conv1.bias, %10, %9, %8, %788, %789, %12, %788, %788, %790): kernel weights has count 9408 but 441 was expected
ERROR: [TRTorch Conversion Context] - %791 : Tensor = aten::_convolution(%x.1, %self.conv1.weight, %self.conv1.bias, %10, %9, %8, %788, %789, %12, %788, %788, %790): count of 9408 weights in kernel, but kernel dimensions (7,7) with 3 input channels, 3 output channels and 1 groups were specified. Expected Weights count is 3 * 7*7 * 3 / 1 = 441
ERROR: [TRTorch Conversion Context] - %791 : Tensor = aten::_convolution(%x.1, %self.conv1.weight, %self.conv1.bias, %10, %9, %8, %788, %789, %12, %788, %788, %790): kernel weights has coun
t 9408 but 441 was expected
ERROR: [TRTorch Conversion Context] - %791 : Tensor = aten::_convolution(%x.1, %self.conv1.weight, %self.conv1.bias, %10, %9, %8, %788, %789, %12, %788, %788, %790): count of 9408 weights i
n kernel, but kernel dimensions (7,7) with 3 input channels, 3 output channels and 1 groups were specified. Expected Weights count is 3 * 7*7 * 3 / 1 = 441
ERROR: [TRTorch Conversion Context] - %791 : Tensor = aten::_convolution(%x.1, %self.conv1.weight, %self.conv1.bias, %10, %9, %8, %788, %789, %12, %788, %788, %790): kernel weights has coun
t 9408 but 441 was expected
ERROR: [TRTorch Conversion Context] - %791 : Tensor = aten::_convolution(%x.1, %self.conv1.weight, %self.conv1.bias, %10, %9, %8, %788, %789, %12, %788, %788, %790): count of 9408 weights i
n kernel, but kernel dimensions (7,7) with 3 input channels, 3 output channels and 1 groups were specified. Expected Weights count is 3 * 7*7 * 3 / 1 = 441
ERROR: [TRTorch Conversion Context] - Parameter check failed at: ../builder/Network.cpp::addScaleNd::737, condition: nbSpatialDims == 2 || nbSpatialDims == 3
Segmentation fault (core dumped)

**Comment by narendasan** [comment]
Hmm, It would be good to get the debug messages here, not sure why you see failures this early in the graph, you can enable them with 
```py
trtorch.logging.set_reportable_log_level(trtorch.logging.Level.Debug)
```

Taking a look at your code, support for the script path for models like ResNet (including ResNet152) just landed in master this week, so I would say try rebuilding with latest master if you want to use `torch.jit.script`. Otherwise you can try the tracing path which is tested to work on ResNet152 from early on in the project. 

I tried your code on master and it seems to work.

**Comment by esghif** [comment]
I can confirm that it is working with newer version of TRTorch.  I was using 223d8dd
